common/flatpak-dir.c

```diff
@@ -16672,19 +16677,22 @@ find_used_refs (FlatpakDir         *self,
  * flatpak_installation_list_unused_refs_with_options().
  * The returned pointer array is transfer full. */
 char **
-flatpak_dir_list_unused_refs (FlatpakDir         *self,
-                              const char         *arch,
-                              GHashTable         *metadata_injection,
-                              GHashTable         *eol_injection,
-                              const char * const *refs_to_exclude,
-                              gboolean            filter_by_eol,
-                              GCancellable       *cancellable,
-                              GError            **error)
+flatpak_dir_list_unused_refs (FlatpakDir            *self,
+                              const char            *arch,
+                              GHashTable            *metadata_injection,
+                              GHashTable            *eol_injection,
+                              const char * const    *refs_to_exclude,
+                              FlatpakDirFilterFlags  filter_flags,
+                              GCancellable          *cancellable,
+                              GError               **error)
 {
   g_autoptr(GHashTable) used_refs = NULL;
+  g_autoptr(GHashTable) autoprune_refs = NULL;
   g_autoptr(GHashTable) excluded_refs_ht = NULL;
   g_autoptr(GPtrArray) refs =  NULL;
   g_autoptr(GPtrArray) runtime_refs = NULL;
+  gboolean filter_by_eol = filter_flags & FLATPAK_DIR_FILTER_EOL;
+  gboolean filter_by_autoprune = filter_flags & FLATPAK_DIR_FILTER_AUTOPRUNE;
```

Comment:
This is wrong: it will assign a value other than `{0, 1}` to a `gboolean`. Better as:

```suggestion
  gboolean filter_by_eol = (filter_flags & FLATPAK_DIR_FILTER_EOL) != 0;
  gboolean filter_by_autoprune = (filter_flags & FLATPAK_DIR_FILTER_AUTOPRUNE) != 0;
```

or equivalently

```suggestion
  gboolean filter_by_eol = !!(filter_flags & FLATPAK_DIR_FILTER_EOL);
  gboolean filter_by_autoprune = !!(filter_flags & FLATPAK_DIR_FILTER_AUTOPRUNE);
```